### ISSUE TITLE
unable to connect to database: 1698, Access denied

### ISSUE THREAD

**Issue by mikegwatson** [mwadman/ansible-librenms#1]
1. I needed to run ```pip3 install pymysql``` prior to running the mwadman.librenms role.

2. This isn't an issue with ansible-librenms play, per se, but when this play is run with an Ubuntu 18.04/19.04 host the following error occurs: **unable to connect to database: (1698, \"Access denied for user 'root'@'localhost'\"**  I believe this is an Ansible bug, but I'm mentioning it here, in case anyone else runs into it. <br><br> The issue can be mitigated, in this role, by adding the line ```login_unix_socket: /var/run/mysqld/mysqld.sock``` to roles/mwadman.librenms/tasks/setup-MariaDB.yml, as shown below: 
```
- name: Create database via local socket
  mysql_db:
    name: "{{ librenms_database_name }}"
    login_host: "{{ librenms_database_login_host }}"
    login_port: "{{ librenms_database_login_port }}"
    login_user: "{{ librenms_database_login_user }}"
    login_password: "{{ librenms_database_login_password }}"
    encoding: utf8
    collation: utf8_unicode_ci
    login_unix_socket: /var/run/mysqld/mysqld.sock
  notify: Restart mysql

- name: Create database user
  mysql_user:
    name: "{{ librenms_database_user }}"
    host: "{{ librenms_database_host }}"
    password: "{{ librenms_database_password }}"
    login_host: "{{ librenms_database_login_host }}"
    login_port: "{{ librenms_database_login_port }}"
    login_user: "{{ librenms_database_login_user }}"
    login_password: "{{ librenms_database_login_password }}"
    priv: '{{ librenms_database_name }}.*:ALL'
    login_unix_socket: /var/run/mysqld/mysqld.sock
  notify: Restart mysql
```
This is further documented [here](https://github.com/ansible/ansible/issues/47736).

**Comment by mwadman** [comment]
Thanks for the feedback Mike, I'll look to add these in to the role for future users.

Regarding `pip3 install pymysql`, was this needed on the control host or the host LibreNMS is being installed to?

**Comment by mikegwatson** [comment]
Hi Michael,

You're welcome.

In my case I needed to run ```pip3 install pymysql``` on the host LibreNMS is being installed to prior to running the mwadman.librenms role.

Cheers,
Mike

**Comment by mwadman** [comment]
Resolved in commits 43222ca (added python-pymysql package) and c4d01a2 (split local and remote host MariaDB set up tasks).
